src/FsCheck/Gen.fs

```diff
@@ -181,7 +181,7 @@ module Gen =
         | :? array<_> as arr ->
             elementsArr arr
         | _ ->
-            choose (0, (Seq.length xs)-1) |> map (flip Seq.nth xs)
+            choose (0, (Seq.length xs)-1) |> map (flip Seq.item xs)
```

Comment:
That means you're compiling against a later version of FSharp.Core. FsCheck on master is compiled against 4.3.1.0 - this was not obsoleted in that version. So there is no reason for the ifdef. FsCheck project indeed has warnings as errors turned on.